### PR TITLE
Persistent subscriptions to $all

### PR DESCRIPTION
**Pull request by pvanbuijtene**
Adds: Support for subscribing to persistent subscriptions to $all
Fixes: Persistent streams to start from the end by default.

**Breaking:** As defined by the _Persistent Subscriptions RFC_, the initial value for a persistent stream has been changed to start from the `end`.

### CODE REVIEW

**Review by YoEight — APPROVED** [review]
LGTM. Don't forget to squash your commits before merging

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        this.maxSubscriberCount = maxSubscriberCount;
+        this.messageTimeoutMs = messageTimeoutMs;
+        this.minCheckpointCount = minCheckpointCount;
+        this.readBatchSize = readBatchSize;
+        this.fromStart = fromStart;
+        this.fromEnd = fromEnd;
+        this.strategy = strategy;
+    }
+
+    public int getCheckpointAfterMs() {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java:40]: **FIX:** Setting does not match the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R51) 
```suggestion
    public int getCheckpointAfter() {
```
Similar comment below: https://github.com/EventStore/EventStoreDB-Client-Java/pull/84/files#diff-b38c322fb2ef3a75dc3289f2babf28c7359937f0530d73c8f3a3734ac18b4dceR20

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        return checkpointAfterMs;
+    }
+
+    public boolean isExtraStatistics() {
+        return extraStatistics;
+    }
+
+    public boolean isResolveLinks() {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java:61]: **FIX:** Setting does not match the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R43). And it's not aligned with other options. See more in e.g. [OptionsWithResolveLinkTosBase](https://github.com/EventStore/EventStoreDB-Client-Java/blob/trunk/db-client-java/src/main/java/com/eventstore/dbclient/OptionsWithResolveLinkTosBase.java#L19)
```suggestion
    public boolean resolveLinkTos() {
```
The same comment applies e.g. https://github.com/EventStore/EventStoreDB-Client-Java/pull/84/files#diff-b38c322fb2ef3a75dc3289f2babf28c7359937f0530d73c8f3a3734ac18b4dceR61.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+
+    public int getHistoryBufferSize() {
+        return historyBufferSize;
+    }
+
+    public int getLiveBufferSize() {
+        return liveBufferSize;
+    }
+
+    public int getMaxCheckpointCount() {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java:90]: **FIX:** Setting does not match the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R43).
```suggestion
    public int getCheckPointUpperBound() {
```
The same comment applies e.g. https://github.com/EventStore/EventStoreDB-Client-Java/pull/84/files#diff-b38c322fb2ef3a75dc3289f2babf28c7359937f0530d73c8f3a3734ac18b4dceR28.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    public int getMaxRetryCount() {
+        return maxRetryCount;
+    }
+
+    public int getMaxSubscriberCount() {
+        return maxSubscriberCount;
+    }
+
+    public int getMessageTimeoutMs() {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java:125]: **FIX:** Setting does not match the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R46).
```suggestion
    public int getMessageTimeout() {
```
The same comment applies e.g. https://github.com/EventStore/EventStoreDB-Client-Java/pull/84/files#diff-b38c322fb2ef3a75dc3289f2babf28c7359937f0530d73c8f3a3734ac18b4dceR25.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        return messageTimeoutMs;
+    }
+
+    public int getMinCheckpointCount() {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java:133]: **FIX:** Setting does not match the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R52).
```suggestion
    public int getCheckPointLowerBound() {
```
The same comment applies e.g. https://github.com/EventStore/EventStoreDB-Client-Java/pull/84/files#diff-b38c322fb2ef3a75dc3289f2babf28c7359937f0530d73c8f3a3734ac18b4dceR27.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+        return minCheckpointCount;
+    }
+
+    public int getReadBatchSize() {
+        return readBatchSize;
+    }
+
+    public boolean getFromStart() {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java]: **CONCERN:** I think it would be more straightforward to introduce enum with the explicit values and have a single getter. Accordingly, for the setters in the builders, we can have a setter with enum plus the syntactic sugar as we have right now. 
Also, shouldn't we allow to set the specific position? 
See more in the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R44).

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+        return fromStart;
+    }
+
+    public boolean getFromEnd() {
+        return fromEnd;
+    }
+
+    public ConsumerStrategy getStrategy() {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java]: **FIX:** Setting does not match the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R55).
```suggestion
    public ConsumerStrategy getConsumerStrategyName() {
```
The same comment applies e.g. https://github.com/EventStore/EventStoreDB-Client-Java/pull/84/files#diff-b38c322fb2ef3a75dc3289f2babf28c7359937f0530d73c8f3a3734ac18b4dceR33.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java [hunk 8]
```diff
@@ -0,0 +1,141 @@
...
+    protected int minCheckpointCount;
+    protected int readBatchSize;
+    protected ConsumerStrategy strategy;
+
+    public AbstractPersistentSubscriptionSettingsBuilder() {
+        checkpointAfterMs = 2_000;
+        resolveLinks = false;
+        extraStatistics = false;
+        fromStart = false;
+        fromEnd = false;
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java]: **FIX:** Based on the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R44) the default starting point should be the end.
```suggestion
        fromEnd = true;
```

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java [hunk 9]
```diff
@@ -0,0 +1,141 @@
...
+        minCheckpointCount = settings.getMinCheckpointCount();
+        maxCheckpointCount = settings.getMaxCheckpointCount();
+        maxSubscriberCount = settings.getMaxSubscriberCount();
+        liveBufferSize = settings.getLiveBufferSize();
+        readBatchSize = settings.getReadBatchSize();
+        historyBufferSize = settings.getHistoryBufferSize();
+        strategy = settings.getStrategy();
+    }
+
+    public T enableLinkResolution() {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java:53]: **FIX:** ResolveLinkTos naming is not aligned with other options. See more e.g. in [OptionsWithResolveLinkTosBase](https://github.com/EventStore/EventStoreDB-Client-Java/blob/trunk/db-client-java/src/main/java/com/eventstore/dbclient/OptionsWithResolveLinkTosBase.java#L19).

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java [hunk 10]
```diff
@@ -0,0 +1,141 @@
+package com.eventstore.dbclient;
+
+public class AbstractPersistentSubscriptionSettingsBuilder<T> {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java:5]: See related comments in [AbstractPersistentSubscriptionSettings](https://github.com/EventStore/EventStoreDB-Client-Java/pull/84/files#diff-3f22e62cf5819afff2ad131e5e7058c61029a3bc78c5490c78d1f057b8ebd9dbR3).

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java [hunk 11]
```diff
@@ -0,0 +1,141 @@
...
+    }
+
+    public T fromEnd() {
+        this.fromEnd = true;
+        this.fromStart = false;
+        return (T) this;
+    }
+
+
+    public T historyBufferSize(int value) {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java:136]: **SUGGESTION:** I think that typically in Java, such setters should start with the prefix `set` or the name should be an imperative form.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractUpdatePersistentSubscription.java [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+                    .setMaxCheckpointCount(settings.getMaxCheckpointCount())
+                    .setMessageTimeoutMs(settings.getMessageTimeoutMs())
+                    .setMaxSubscriberCount(settings.getMaxSubscriberCount())
+                    .setMaxRetryCount(settings.getMaxRetryCount())
+                    .setLiveBufferSize(settings.getLiveBufferSize())
+                    .setHistoryBufferSize(settings.getHistoryBufferSize())
+                    .setExtraStatistics(settings.isExtraStatistics())
+                    .setCheckpointAfterMs(settings.getCheckpointAfterMs());
+
+            switch (settings.getStrategy()) {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/AbstractUpdatePersistentSubscription.java]: **CONCERN:** Aren't we missing `PinnedByCorrelation` here? See more in the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R71).

**Comment by oskardudycz** [comment]
@pvanbuijtene, @hayley-jean I'm not sure if that's out of the scope of this PR, but I don't see the implementation of:
- the `AutoAck` [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R62),
- `ReplayParkedMessages` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R25),
- `RestartPersistentSubscriptionSubsystem` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R25),
- `GetInfo` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R26),
- `ListAll` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R29),
- `ListForStream`method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R30).
- `PinnedByCorrelation` strategy mentioned here: https://github.com/EventStore/EventStoreDB-Client-Java/pull/84#discussion_r678078940.

@hayley-jean, could you wage what should go here and what's not? 

I guess that probably `AutoAck` should go in this one. The rest can go in the dedicated, follow-up PRs.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java (hunk 1 above)
Comment (hayley-jean) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java:40]: We could also have an overload that takes a duration without the `Ms` in the name

File: db-client-java/src/test/java/com/eventstore/dbclient/DeletePersistentSubscriptionTests.java [hunk 13]
```diff
@@ -15,4 +11,13 @@ public void testDeletePersistentSub() throws Throwable {
...
     }
+
+    @Test
+    public void testDeletePersistentSubToAll() throws Throwable {
+        client.createToAll("aGroupUpd")
+                .get();
+
+        client.deleteToAll("aGroupUpd")
+                .get();
+    }
```
Comment (oskardudycz) [inline db-client-java/src/test/java/com/eventstore/dbclient/DeletePersistentSubscriptionTests.java:22]: **CONCERN:** Shouldn't we have some assertion that verifies if the operations succeeded? We can have the false-positive test results if delete silently fails. Similar comment to other added/changed tests.

File: db-client-java/src/test/java/com/eventstore/dbclient/SubscribePersistentSubcription.java [hunk 14]
```diff
@@ -82,4 +82,59 @@ public void onCancelled(PersistentSubscription subscription) {
 
         Assert.assertEquals(6, result.get().intValue());
     }
+
```
Comment (oskardudycz) [inline db-client-java/src/test/java/com/eventstore/dbclient/SubscribePersistentSubcription.java:85]: **SUGGESTION:** Can we add test scenarios for the negative cases, e.g.:
- what happens if subscribing to a not available database,
- nodes were disconnected during the persistent subscription liveness,
- error while handling the persistent subscription logic,
- etc.

File: db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBPersistentSubscriptionsClient.java [hunk 15]
```diff
@@ -11,20 +11,52 @@ public static EventStoreDBPersistentSubscriptionsClient create(EventStoreDBClien
...
     public CompletableFuture create(String stream, String group) {
-        return this.create(stream, group, CreatePersistentSubscriptionOptions.get());
+        return this.createToStream(stream, group);
     }
 
+    public CompletableFuture createToAll(String group) {
+        return this.createToAll(group, CreatePersistentSubscriptionToAllOptions.get());
+    }
+
+    public CompletableFuture createToStream(String stream, String group) {
```
Comment (oskardudycz) [inline db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBPersistentSubscriptionsClient.java]: **CONCERN:** Based on the [RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R18), this method is redundant, as we can just use `create`. The same applies to other `ToStream` management methods.

**Review by oskardudycz — CHANGES_REQUESTED** [review]
See the comments above.

**Comment by oskardudycz** [comment]
Also, documentation snippets are missing. See the .NET snippets for persistent subscriptions: https://github.com/EventStore/EventStore-Client-Dotnet/blob/master/samples/persistent-subscriptions/Program.cs.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java (hunk 7 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java]: @oskardudycz `getConsumerStrategyName` suggests a string is being returned, I know it's the name from the RFC but would `getConsumerStrategy` not be better since an enum is returned ?

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java (hunk 11 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java:136]: Agree, but in this case it's a builder which normally has a more fluent syntax?

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java (hunk 9 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java:53]: This was already in-place, deprecated it and linked to new `resolveLinkTos()`.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java (hunk 4 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java:125]: Added an overload which returns the `Duration`.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java (hunk 6 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java]: I refactored the `fromStart` and `fromEnd`, the implementations can be found in the specific classes. The position can be set by the builder and with the constructor.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java (hunk 8 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java]: The default for streams is already defined to begin from the start, do we want to change this?
I changed it for the `all` stream subscription.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java (hunk 9 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java:53]: This was already in-place, deprecated it and linked to new method.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java (hunk 11 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java:97]: Agree, but this is a builder class which normally has a more fluent naming?

File: db-client-java/src/test/java/com/eventstore/dbclient/DeletePersistentSubscriptionTests.java (hunk 13 above)
Comment (pvanbuijtene) [reply db-client-java/src/test/java/com/eventstore/dbclient/DeletePersistentSubscriptionTests.java:22]: This is how the current tests are written, currently the `List`/`Get` operations aren't implemented yet for persistent subscriptions. I would suggest to implement this in separate PR and modify/add the tests.

File: db-client-java/src/test/java/com/eventstore/dbclient/SubscribePersistentSubcription.java (hunk 14 above)
Comment (pvanbuijtene) [reply db-client-java/src/test/java/com/eventstore/dbclient/SubscribePersistentSubcription.java:85]: Agree, see previous comment. https://github.com/EventStore/EventStoreDB-Client-Java/pull/84#discussion_r678663827

File: db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBPersistentSubscriptionsClient.java (hunk 15 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/EventStoreDBPersistentSubscriptionsClient.java]: Removed `toStream` classes and methods.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractUpdatePersistentSubscription.java (hunk 12 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractUpdatePersistentSubscription.java]: Think this is not implemented yet, can't find this on the server ?

**Comment by pvanbuijtene** [comment]
> @pvanbuijtene, @hayley-jean I'm not sure if that's out of the scope of this PR, but I don't see the implementation of:
> 
> * the `AutoAck` [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R62),
> * `ReplayParkedMessages` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R25),
> * `RestartPersistentSubscriptionSubsystem` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R25),
> * `GetInfo` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R26),
> * `ListAll` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R29),
> * `ListForStream`method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R30).
> * `PinnedByCorrelation` strategy mentioned here: [Persistent subscriptions to $all #84 (comment)](https://github.com/EventStore/EventStoreDB-Client-Java/pull/84#discussion_r678078940).
> 
> @hayley-jean, could you wage what should go here and what's not?
> 
> I guess that probably `AutoAck` should go in this one. The rest can go in the dedicated, follow-up PRs.

`AutoAck` is also not implemented for normal persistent subscriptions, suggest to do this in separate PR.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java (hunk 7 above)
Comment (hayley-jean) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java]: I think if it's an enum being returned `getConsumerStrategy` makes the most sense

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractUpdatePersistentSubscription.java (hunk 12 above)
Comment (hayley-jean) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractUpdatePersistentSubscription.java]: PinnedByCorrelation isn't available on the server proto yet, so should come in a later PR

**Comment by hayley-jean** [comment]
> @pvanbuijtene, @hayley-jean I'm not sure if that's out of the scope of this PR, but I don't see the implementation of:
> 
> * the `AutoAck` [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R62),
> * `ReplayParkedMessages` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R25),
> * `RestartPersistentSubscriptionSubsystem` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R25),
> * `GetInfo` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R26),
> * `ListAll` method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R29),
> * `ListForStream`method [mentioned in the RFC](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R30).
> * `PinnedByCorrelation` strategy mentioned here: [Persistent subscriptions to $all #84 (comment)](https://github.com/EventStore/EventStoreDB-Client-Java/pull/84#discussion_r678078940).
> 
> @hayley-jean, could you wage what should go here and what's not?
> 
> I guess that probably `AutoAck` should go in this one. The rest can go in the dedicated, follow-up PRs.

The missing operations still need to be added to the server proto before we can implement them (see [here](https://github.com/EventStore/home/pull/550/files#diff-ad3a37ec8a5416ddcd905109248d13dba6a72a352ce7ee41b9d3d004aae1ff95R75))

AutoAck could go in this PR or in a follow-up one that also implements it for the other persistent subscriptions.

**Review by YoEight — APPROVED** [review]

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java (hunk 1 above)
Comment (oskardudycz) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettings.java:40]: I think that having both methods is confusing. The general convention is to not include `Ms` in the name as all our timeouts should be set in milliseconds.

**Comment by oskardudycz** [comment]
@hayley-jean, I think that we should add an issue to remove all deprecated fields in the next major release.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java (hunk 8 above)
Comment (oskardudycz) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java]: I think that we should be aligned with the RFC also for the starting stream position. It will be a breaking change.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java (hunk 11 above)
Comment (oskardudycz) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java:136]: It is not the hill I'd die for, but IMHO, having set in this case would be more explicit in this case. But it's not a blocker for me to keep it as it is.

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractUpdatePersistentSubscription.java (hunk 12 above)
Comment (oskardudycz) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractUpdatePersistentSubscription.java]: It might be worth creating an issue for that to not forget about it.

File: db-client-java/src/test/java/com/eventstore/dbclient/DeletePersistentSubscriptionTests.java (hunk 13 above)
Comment (oskardudycz) [reply db-client-java/src/test/java/com/eventstore/dbclient/DeletePersistentSubscriptionTests.java:22]: I'm good with adding them in the separate PR, but it'd be good to create an issue for that to not forget about it.

**Review by oskardudycz — APPROVED** [review]
@pvanbuijtene, thanks for the fixes. Approving 👍 

It'd be good to clarify https://github.com/EventStore/EventStoreDB-Client-Java/pull/84#discussion_r682376481 before merging.

File: db-client-java/src/test/java/com/eventstore/dbclient/DeletePersistentSubscriptionTests.java (hunk 13 above)
Comment (pvanbuijtene) [reply db-client-java/src/test/java/com/eventstore/dbclient/DeletePersistentSubscriptionTests.java:22]: Done https://github.com/EventStore/EventStoreDB-Client-Java/issues/87

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractUpdatePersistentSubscription.java (hunk 12 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractUpdatePersistentSubscription.java]: Done https://github.com/EventStore/EventStoreDB-Client-Java/issues/88

File: db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java (hunk 8 above)
Comment (pvanbuijtene) [reply db-client-java/src/main/java/com/eventstore/dbclient/AbstractPersistentSubscriptionSettingsBuilder.java]: Done.

**Comment by pvanbuijtene** [comment]
**Issues created:**
Remove deprecated methods and classes https://github.com/EventStore/EventStoreDB-Client-Java/issues/86
Improve code coverage for persistent subscriptions https://github.com/EventStore/EventStoreDB-Client-Java/issues/87
Support PinnedByCorrelation strategy for persistent subscriptions https://github.com/EventStore/EventStoreDB-Client-Java/issues/88
